source/mongodb-handshake/handshake.rst

```diff
@@ -469,16 +469,12 @@ The entire metadata BSON document MUST NOT exceed 512 bytes. This includes all
 BSON overhead.  The ``client.application.name`` cannot exceed 128 bytes.  MongoDB
 will return an error if these limits are not adhered to, which will result in
 handshake failure. Drivers MUST validate these values and truncate or omit driver
-provided values if necessary.  Implementors SHOULD prioritize fields to preserve in
-this order:
+provided values if necessary.  Implementors SHOULD cumulatively update fields in
+the following order until the document is under the size limit:
 
-1. ``application.name``
-2. ``driver.*``
-3. ``os.type``
-4. ``env.name``
-5. ``os.*`` (except ``type``)
-6. ``env.*`` (except ``name``)
-7. ``platform``
+1. Truncate ``platform``.
+2. Omit fields from ``env`` except ``env.name``.
+3. Omit the ``env`` document entirely.
 
```

Comment:
```
/* REQUIRED */
        os: {
            type: "<string>",         /* REQUIRED */
            name: "<string>",         /* OPTIONAL */
            architecture: "<string>", /* OPTIONAL */
            version: "<string>"       /* OPTIONAL */
        },
```
Since the spec indicates the `os` field only requires `os.type` would it make sense to also include language such as: _Omit fields from `os` except `os.type`._?

Reply:
Also due to the need for data in the `platform` field (for Node.js, and likely Java) I'd advocate for reorganizing this to be:

1. Omit fields from ``env`` except ``env.name``.
2. Omit fields from os except os.type
3. Omit the ``env`` document entirely.
4. Truncate platform

Reply:
Sounds reasonable - I've updated to this.

Reply:
What happens if the client metadata exceeds the 512 byte limit with os.type but is under the limit without it? IIUC this is technically possible since the application name is configurable.

Reply:
That's orthogonal to this PR, and is covered under the more general
> Drivers MUST validate these values and truncate or omit driver provided values if necessary.
The list here is giving relative priorities of some fields but it's neither mandatory nor exhaustive.